### PR TITLE
Add Python 3.14 support

### PR DESCRIPTION
**Pull request by andersonhc**
Add Python 3.14 support

**Checklist**:

- [ ] A unit test is covering the code added / modified by this PR

- [ ] In case of a new feature, docstrings have been added, with also some documentation in the `docs/` folder

- [x] A mention of the change is present in `CHANGELOG.md`

- [ ] This PR is ready to be merged 

By submitting this pull request, I confirm that my contribution is made under the terms of the [GNU LGPL 3.0 license](https://github.com/py-pdf/fpdf2/blob/master/LICENSE).

### CODE REVIEW

File: test/image/test_image_info.py [hunk 1]
```diff
@@ -38,18 +39,23 @@ def test_get_img_info():
             info = fpdf.image_parsing.get_img_info(blob)
             short_info = {}
             for k, v in info.items():
+                if k == "smask":
+                    short_info["smask"] = zlib.decompress(v)
                 if k in short_keys:
```
Comment (Lucas-C) [inline test/image/test_image_info.py]:
```suggestion
                if k == "smask":
                    short_info["smask"] = zlib.decompress(v)
                elif k in short_keys:
```

Comment (Lucas-C) [reply test/image/test_image_info.py]: Why is this change regarding `smask` needed?

File: .github/workflows/continuous-integration-workflow.yml [hunk 2]
```diff
@@ -115,19 +115,45 @@ jobs:
...
+            : > "$CONSTRAINTS_FILE"   # create empty file
+            echo "Created empty $CONSTRAINTS_FILE"
+          fi
+          # --- Install dependencies respecting constraints ---
+          pip install --upgrade . -r test/requirements.txt -c "$CONSTRAINTS_FILE"
       - name: Run tests ☑
+        shell: bash
         env:
           CHECK_EXEC_TIME: ${{ matrix.python-version == '3.10' && 'test-enabled' || '' }}
           CHECK_RSS_MEMORY: ${{ matrix.python-version == '3.13' && 'test-enabled' || '' }}
```
Comment (Lucas-C) [inline .github/workflows/continuous-integration-workflow.yml]:
```suggestion
          CHECK_RSS_MEMORY: ${{ matrix.python-version == '3.14' && 'test-enabled' || '' }}
```

File: .github/workflows/continuous-integration-workflow.yml [hunk 3]
```diff
@@ -115,19 +115,45 @@ jobs:
...
+          CONSTRAINTS_FILE="constraints.txt"
+          if [[ "${{ matrix.python-version }}" == "3.14" ]]; then
+            echo "numpy>=2.3.3" > "$CONSTRAINTS_FILE"
+            echo "Added numpy>=2.3.3 to $CONSTRAINTS_FILE"
+          else
+            : > "$CONSTRAINTS_FILE"   # create empty file
+            echo "Created empty $CONSTRAINTS_FILE"
+          fi
+          # --- Install dependencies respecting constraints ---
+          pip install --upgrade . -r test/requirements.txt -c "$CONSTRAINTS_FILE"
```
Comment (Lucas-C) [inline .github/workflows/continuous-integration-workflow.yml:142]: Why adding `numpy>=2.3.3` there?
Why is it needed to introduce a `constraints.txt` file?

Wouldn't adding `numpy>=2.3.3` to `test/requirements.txt` be enough?

Maybe that's worth an explanatory comment 🙂

**Review by Lucas-C — APPROVED** [review]

**Comment by andersonhc** [comment]
@Lucas-C 
The Windows build of cpython 3.14 is now using zlib-ng instead of the standard zlib
https://github.com/python/cpython/issues/91349
https://github.com/python/cpython/pull/131438

This change alters the output of zlib.compress, which means that our PDF file comparisons no longer match byte-for-byte between platforms.
The generated compressed streams differ even though the decompressed content is identical.

To mitigate this problem for now I:
- Changed our github workflow to make sure `qpdf` is used for the Windows tests so we compare uncompressed data
- There were 3 edge cases remaining because of compressed data embedded on PDF files that `qpdf` won't decompress - I am skipping on Python 3.14+Windows for now

I believe this deserves more thought and a long term solution, but I'm planning on a new release in the next few weeks and I'd like to have 3.14 wheels published.

**Comment by Lucas-C** [comment]
>The Windows build of cpython 3.14 is now using zlib-ng instead of the standard zlib [python/cpython#91349](https://github.com/python/cpython/issues/91349) [python/cpython#131438](https://github.com/python/cpython/pull/131438)
> [...]
> This change alters the output of zlib.compress, which means that our PDF file comparisons no longer match byte-for-byte between platforms. > I believe this deserves more thought and a long term solution, but I'm planning on a new release in the next few weeks and I'd like to have 3.14 wheels published.

Alright.

So far we did not bother running tests with `qpdf` under Windows,
I just wonder if it's worth the hassle, and to introduce minor kludges add this?

Just some humble questioning, I approved the PR and I'm fine if you decide to merge it with those additions 🙂

File: .github/workflows/continuous-integration-workflow.yml (hunk 3 above)
Comment (andersonhc) [reply .github/workflows/continuous-integration-workflow.yml:142]: I fought pip for a day and couldn't find another solution... It was installing 2.3.3 and then downgrading back to numpy 2.2.6. I believe the problem is a dependency downstream from camelot-py.
I added the constraints file on python 3.14 builds only and skipped the camelot table extraction test on 3.14.
When camelot publish a Python 3.14 compatible build we should be OK to remove this.

I'll add comments to have it documented.

File: test/image/test_image_info.py (hunk 1 above)
Comment (andersonhc) [reply test/image/test_image_info.py]: zlib.compress() is producing different results on Python 3.14 windows and failing if we compare compressed data.
I changed to compare the decompressed streams.
